### ISSUE TITLE
lingua tries to use Babel extractors even if Babel is not installed

### ISSUE THREAD

**Issue by sinoroc** [wichert/lingua#94]
It appears that lingua tries to use [Babel](https://pypi.org/project/Babel/) message extractors even if Babel is not installed. For example [Mako](https://pypi.org/project/Mako/) contains custom message extractors for Babel. If Mako is installed but not Babel, lingua recognizes the _Mako-for-Babel_ message extractors, loads them, and finally fails while tryin to use them.

Two things:

1. Mako should declare Babel as an extra dependency alongside the entry points for its custom message extractors
1. lingua should not load entry points when the extra dependencies are missing

See [setuptools documentation on "Dynamic Discovery of Services and Plugins"](https://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins) for details.

**Comment by sinoroc** [comment]
I am working on fixes for both Mako and lingua.

**Comment by sinoroc** [comment]
This is the trace when Mako is installed but not Babel:

```
Traceback (most recent call last):
  File "/home/user/workspace/project/.tox/develop/bin/pot-create", line 11, in <module>
    sys.exit(main())
  File "/home/user/workspace/project/.tox/develop/lib/python3.6/site-packages/lingua/extract.py", line 308, in main
    register_babel_plugins()
  File "/home/user/workspace/project/.tox/develop/lib/python3.6/site-packages/lingua/extractors/babel.py", line 51, in register_babel_plugins
    extractor = entry_point.load(require=True)
  File "/home/user/workspace/project/.tox/develop/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2324, in load
    return self.resolve()
  File "/home/user/workspace/project/.tox/develop/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2330, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/user/workspace/project/.tox/develop/lib/python3.6/site-packages/mako/ext/babelplugin.py", line 8, in <module>
    from babel.messages.extract import extract_python
ModuleNotFoundError: No module named 'babel'
```

**Comment by wichert** [comment]
Fixed by #97
